packages/git/src/browser/git-decorator.ts

```diff
@@ -77,17 +77,11 @@ export class GitDecorator implements TreeDecorator {
             }
         };
         const markers = this.appendContainerChanges(tree, status.changes);
-        processNode(tree.root);
-        const itr = new TreeNodeIterator(tree.root);
-        let node = itr.next();
-        while (!node.done) {
-            processNode(node.value);
-            node = itr.next();
-        }
+        [...new PreOrderTreeIterator(tree.root)].forEach(processNode);
```

Comment:
Does not it iterate twice, first to create an array and the second time to spread it? It is fine to inline `processNode` and use `for`:
```ts
for (const { id } of new PreOrderTreeIterator(tree.root)) {
    const marker = markers.get(id);
    if (marker) {
        result.set(id, marker);
    }
}
```